boxsdk/object/item.py

```diff
@@ -398,3 +399,133 @@ def remove_from_collection(self, collection):
+            List of fields to request.
+        :type fields:
+            `Iterable` of `unicode`
+        """
+        url = self._session.get_url('collaborations')
+        body = {
+            'item': {
+                'type': self.object_type,
+                'id': self.object_id,
+            },
+            'accessible_by': {
+                'type': accessible_by.object_type,
+                'id': accessible_by.object_id,
+            },
+            'role': role,
+        }
+        if can_view_path is not None:
+            body['can_view_path'] = can_view_path
+        params = {}
+        if fields is not None:
+            params['fields'] = ','.join(fields)
+        if notify is not None:
+            params['notify'] = ','.join(fields)
+        response = self._session.post(url, data=json.dumps(body), params=params).json()
+        return self.translator.translate(response['type'])(
+            session=self._session,
+            object_id=response['id'],
+            response_object=response,
+        )
+
+    def collaborate_with_login(self, login, role, can_view_path=None, notify=None, fields=None):
+        """Collaborate user or group onto a Box item.
+
+        :param login:
+            The email address of the person to grant access to.
+        :type login:
+            `unicode`
+        :param role:
+            The permission level to grant the collaborator.
+        :type role:
+            `unicode`
+        :param can_view_path:
+            Indicates whether the user can view the path of the folder collaborated into.
+        :type can_view_path:
+            `bool` or None
+        :param notify:
+            Determines if the collaborator should receive a notification for the collaboration.
+        :type notify:
+            `bool` or None
+        :param fields:
+            List of fields to request.
+        :type fields:
+            `Iterable` of `unicode`
+        """
+        url = self._session.get_url('collaborations')
+        body = {
+            'item': {
+                'type': self.object_type,
+                'id': self.object_id,
+            },
+            'accessible_by': {
+                'type': 'user',
+                'login': login,
+            },
+            'role': role,
+        }
+        if can_view_path is not None:
+            body['can_view_path'] = can_view_path
+        params = {}
+        if fields is not None:
+            params['fields'] = ','.join(fields)
+        if notify is not None:
+            params['notify'] = ','.join(fields)
+        response = self._session.post(url, data=json.dumps(body), params=params).json()
+        return self.translator.translate(response['type'])(
+            session=self._session,
+            object_id=response['id'],
+            response_object=response,
+        )
+
+    def collaborations(self, limit=None, marker=None, fields=None):
+        """
+        Get the entries in the collaboration using marker-based paging.
```

Comment:
We can just remove the part about marker-based paging form this description; that's an implementation detail that the user probably doesn't care about.